src/main/java/com/javadiscord/javabot/commands/reaction_roles/ReactionRoles.java

```diff
@@ -1,152 +1,85 @@
 package com.javadiscord.javabot.commands.reaction_roles;
 
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
 import com.javadiscord.javabot.commands.Responses;
 import com.javadiscord.javabot.commands.SlashCommandHandler;
 import com.javadiscord.javabot.other.Constants;
 import com.javadiscord.javabot.other.Misc;
-import com.mongodb.BasicDBObject;
-import com.mongodb.client.MongoCollection;
-import com.mongodb.client.MongoCursor;
-import com.mongodb.client.MongoDatabase;
 import net.dv8tion.jda.api.EmbedBuilder;
 import net.dv8tion.jda.api.entities.Emoji;
-import net.dv8tion.jda.api.entities.Message;
-import net.dv8tion.jda.api.entities.MessageChannel;
 import net.dv8tion.jda.api.entities.Role;
 import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
 import net.dv8tion.jda.api.interactions.components.Button;
 import net.dv8tion.jda.api.interactions.components.ButtonStyle;
 import net.dv8tion.jda.api.requests.restaction.interactions.ReplyAction;
-import org.bson.Document;
 
+import java.time.Instant;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 
-import static com.javadiscord.javabot.events.Startup.mongoClient;
-import static com.mongodb.client.model.Filters.eq;
-
-// TODO: Redo implementation of this to be much cleaner.
-@Deprecated(forRemoval = true)
 public class ReactionRoles implements SlashCommandHandler {
+
     @Override
     public ReplyAction handle(SlashCommandEvent event) {
-        return switch (event.getSubcommandName()) {
-            case "create" -> create(event,
-                    event.getOption("channel").getAsMessageChannel(),
-                    event.getOption("messageid").getAsString(),
-                    event.getOption("emote").getAsString(),
-                    event.getOption("button-label").getAsString(),
-                    event.getOption("role").getAsRole());
-            case "delete" -> delete(event,
-                    event.getOption("messageid").getAsString(),
-                    event.getOption("button-label").getAsString(),
-                    event.getOption("emote").getAsString());
-            default -> Responses.warning(event, "Unknown subcommand.");
-        };
+        try {
+            return switch (event.getSubcommandName()) {
+                case "create" -> createReactionRole(event);
+                case "delete" -> deleteReactionRole(event);
+                default -> Responses.warning(event, "Unknown subcommand.");
+            };
+        } catch (Exception e) { return Responses.error(event, "```" + e.getMessage() + "```"); }
```

Comment:
It would be a lot simpler if you just made `ReactionRoles extends DelegatingCommandHandler` and override the constructor to define the possible subcommands, instead of doing that manually here. That way the behavior and warning messages are the same for all commands.